crates/ark/src/lsp/symbols.rs

```diff
@@ -414,26 +532,41 @@ fn collect_call_arguments(
                 return Ok(());
             };
 
-            // Recurse into arguments. They might be a braced list, another call
-            // that might contain functions, etc.
-            collect_symbols(ctx, &arg_value, contents, 0, symbols)?;
-
+            // If this is a named function, collect it as a method (new node in the tree)
             if arg_value.kind() == "function_definition" {
                 if let Some(arg_fun) = child.child_by_field_name("name") {
-                    // If this is a named function, collect it as a method
                     collect_method(ctx, &arg_fun, &arg_value, contents, symbols)?;
-                } else {
-                    // Otherwise, just recurse into the function
-                    let body = arg_value.child_by_field_name("body").into_result()?;
-                    collect_symbols(ctx, &body, contents, 0, symbols)?;
+                    return Ok(());
                 };
+                // else fallthrough
             }
 
+            collect_symbols(ctx, &arg_value, contents, 0, symbols)?;
+
             Ok(())
         },
     )
 }
 
+fn collect_function_parameters(
+    ctx: &mut CollectContext,
+    node: &Node,
+    contents: &Rope,
+    symbols: &mut Vec<DocumentSymbol>,
+) -> anyhow::Result<()> {
+    collect_sections(
+        ctx,
+        &node,
+        contents,
+        0,
```

Comment:
Hmmmmmm `0`? Are you sure you shouldn't pass the `current_level` through to here? You do for the function body. Add a test either way about this behavior?

Reply:
Or at least a comment here about why its 0